qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/utils/CDIUtils.java

```diff
@@ -63,4 +73,18 @@ public static String getSimpleName(String javaElementName, String annotationName
 		}
 		return javaElementName;
 	}
+
+	public static boolean isValidBean(IJavaElement javaElement) {
+		try {
+			return (javaElement.getClass().getDeclaringClass() == null
+					&& !Modifier.isAbstract(javaElement.getClass().getModifiers())
+					&& !AnnotationUtils.hasAnnotation((IAnnotatable) javaElement, JAVAX_DECORATOR_ANNOTATION)
+					&& !AnnotationUtils.hasAnnotation((IAnnotatable) javaElement, JAVAX_INJECT_VETOED_ANNOTATION)
+					&& ((IType) javaElement).isClass());
+		} catch (SecurityException e) {
+			return false;
+		} catch (JavaModelException e) {
+			return false;
+		}

```

Comment:
You could probably merge this into just `Exception e` and have it return `false`.